versioned_docs/version-7.x/testing.md

```diff
@@ -54,53 +48,670 @@ Then we need to use this setup file in our jest config. You can add it under `se
+  const navigation = createNavigationContainerRef();
+  render(<DrawerNavigation ref={navigation} />);
+
+  expect(renderCounter).toBe(0);
+
+  // navigate.preload causes React state updates
+  // So it should be wrapped into act
+  act(() => navigation.preload('Settings'));
+
+  expect(screen.queryByText('Profile Screen')).toBeOnTheScreen();
+  expect(screen.queryByText('Settings Screen')).not.toBeOnTheScreen();
+  expect(renderCounter).toBe(1);
+
+  fireEvent.press(screen.queryByText('Navigate to Settings'));
+
+  expect(screen.queryByText('Profile Screen')).not.toBeOnTheScreen();
+  expect(screen.queryByText('Settings Screen')).toBeOnTheScreen();
+  expect(renderCounter).toBe(1);
+});
+```
+
+</TabItem>
+<TabItem value="dynamic" label="Dynamic">
+
+```js
+import { expect, test } from '@jest/globals';
+import { createDrawerNavigator } from '@react-navigation/drawer';
+import {
+  createNavigationContainerRef,
+  NavigationContainer,
+} from '@react-navigation/native';
+import { act, fireEvent, render, screen } from '@testing-library/react-native';
+import { Button, Text, View } from 'react-native';
+
+const ProfileScreen = ({ navigation }) => {
+  return (
+    <View>
+      <Text>Profile Screen</Text>
+      <Button
+        onPress={() => navigation.navigate('Settings')}
+        title="Navigate to Settings"
+      />
+    </View>
+  );
+};
+
+let renderCounter = 0;
+
+const SettingsScreen = () => {
+  renderCounter++;
+  return (
+    <View>
+      <Text>Settings Screen</Text>
+    </View>
+  );
+};
+
+const Drawer = createDrawerNavigator();
+
+const DrawerNavigation = () => {
+  return (
+    <Drawer.Navigator>
+      <Drawer.Screen name="Profile" component={ProfileScreen} />
+      <Drawer.Screen name="Settings" component={SettingsScreen} />
+    </Drawer.Navigator>
+  );
+};
+
+test('navigates to settings with previous preload', () => {
+  const navigation = createNavigationContainerRef();
+  render(
+    <NavigationContainer ref={navigation}>
+      <DrawerNavigation />
+    </NavigationContainer>
+  );
+
+  expect(renderCounter).toBe(0);
+
+  // navigation.preload causes React state updates
+  // So it should be wrapped into act
+  act(() => navigation.preload('Settings'));
+
+  expect(screen.queryByText('Profile Screen')).toBeOnTheScreen();
+  expect(screen.queryByText('Settings Screen')).not.toBeOnTheScreen();
+  expect(renderCounter).toBe(1);
+
+  fireEvent.press(screen.queryByText('Navigate to Settings'));
+
+  expect(screen.queryByText('Profile Screen')).not.toBeOnTheScreen();
+  expect(screen.queryByText('Settings Screen')).toBeOnTheScreen();
+  expect(renderCounter).toBe(1);
+});
+```
+
+</TabItem>
+</Tabs>
+
+For writing tests that include times functions we will have to use [Fake Timers](https://jestjs.io/docs/timer-mocks). They will replace times function implementation to use time from the fake clock.
+
+Let's add another button to the Profile screen, which uses `setTimeout`:
```

Comment:
If we decide on leaving preload test as it is with navigator definition and test in one file, I will adjust this section text, because it's based on previous navigator